quadtree/src/lib.rs

```diff
@@ -55,11 +55,19 @@ pub struct Node {
 }
 
 impl Node {
-    pub fn root_with_bounding_rect(rect: Rect) -> Self {
-        Node {
-            id: NodeId::root(),
+    pub fn from_node_id_and_root_bounding_rect(id: NodeId, rect: Rect) -> Self {
+        let mut ids = vec![id];
+        while let Some(parent_id) = ids.last().and_then(|last| last.parent_id()) {
+            ids.push(parent_id);
+        }
+        let mut node = Self {
+            id: ids.pop().unwrap(),
```

Comment:
Just making sure I understand: at this point `ids.pop().unwrap()` returns the root node, or`id: NodeId::root()`?

Reply:
Yes, I implemented it this way that it also works when just passing the actual root node.